core/transactor/transactor.go

```diff
@@ -0,0 +1,195 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package transactor
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/ethereum/go-ethereum/common"
+	pc "github.com/mysteriumnetwork/payments/crypto"
+	"github.com/mysteriumnetwork/payments/registration"
+	"github.com/pkg/errors"
+
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/node/metadata"
+	"github.com/mysteriumnetwork/node/requests"
+)
+
+type transactor struct {
+	http            requests.HTTPTransport
+	endpointAddress string
+	signerFactory   identity.SignerFactory
+	registryAddress string
+	accountantID    string
+}
+
+// NewTransactor creates and returns new Transactor instance
+func NewTransactor(bindAddress, endpointAddress, registryAddress, accountantID string, signerFactory identity.SignerFactory) *transactor {
+	return &transactor{
+		http:            requests.NewHTTPClient(bindAddress, 20*time.Second),
+		endpointAddress: endpointAddress,
+		signerFactory:   signerFactory,
+		registryAddress: registryAddress,
+		accountantID:    accountantID,
+	}
+}
+
+// Fees represents fees applied by Transactor
+// swagger:model Fees
+type Fees struct {
+	Transaction  uint64 `json:"transaction"`
+	Registration uint64 `json:"registration"`
+}
+
+// IdentityRegistrationRequestDTO represents the identity registration user input parameters
+// swagger:model IdentityRegistrationRequestDTO
+type IdentityRegistrationRequestDTO struct {
+	// Stake is used by Provider, default 0
+	Stake uint64 `json:"stake,omitempty"`
+	// Cache out address for Provider
+	Beneficiary string `json:"beneficiary,omitempty"`
+	// Fee: negotiated fee with transactor
+	Fee uint64 `json:"fee,omitempty"`
+}
+
+// IdentityRegistrationRequest represents the identity registration request body
+type IdentityRegistrationRequest struct {
+	RegistryAddress string `json:"registryAddress"`
+	AccountantID    string `json:"accountantID"`
+	// Stake is used by Provider, default 0
+	Stake uint64 `json:"stake"`
+	// Fee: negotiated fee with transactor
+	Fee uint64 `json:"fee"`
+	// Beneficiary: Provider channelID by default, optionally set during Identity registration.
+	// Can be updated later through transactor. We can check it's value directly from SC.
+	// Its a cache out address
+	Beneficiary string `json:"beneficiary"`
+	// Signature from fields above
+	Signature string `json:"signature"`
+	Identity  string `json:"identity"`
+}
+
+//  FetchFees fetches current transactor fees
+func (t *transactor) FetchFees() (Fees, error) {
+	f := Fees{}
+
+	req, err := requests.NewGetRequest(t.endpointAddress, "fee/register", nil)
+	if err != nil {
+		return f, errors.Wrap(err, "failed to fetch transactor fees")
+	}
+
+	err = t.http.DoRequestAndParseResponse(req, &f)
+	if err != nil {
```

Comment:
We don't need this if block, just:
```suggestion
	return f, err
```